### ISSUE TITLE
`dotnet restore` leaves temporary folders behind

### ISSUE THREAD

**Issue by moozzyk** [dotnet/cli#811]
Each time a project is restored a temporary folder is being created and left behind (empty) e.g (notice the guid folders):

```
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        1/12/2016   1:38 PM                2ddb1902-b1bc-4289-9082-6650ac75b820
d-----        1/12/2016   1:20 PM                73777215-e8cd-44a7-a66e-eb3904b337be
d-----        1/12/2016  11:07 AM                artifacts
d-----        1/12/2016   1:20 PM                bin
d-----        1/12/2016   1:39 PM                e37f6145-7b60-478d-98f6-4ca07787c0e6
d-----        1/12/2016   1:39 PM                e809ef58-0d3f-4ae5-bd73-134fa9707746
d-----        1/12/2016   1:38 PM                ec198ccd-574e-4c85-8a38-81bc0521bfd5
d-----        1/12/2016   1:20 PM                f63289d6-26b4-4b6e-a4e5-ad31cb834261
d-----        1/12/2016   1:20 PM                obj
d-----        1/12/2016  10:47 AM                Properties
d-----        1/12/2016  10:47 AM                wwwroot
-a----        1/12/2016   1:11 PM            743 project.json
-a----        1/12/2016   1:39 PM         436043 project.lock.json
-a----        10/7/2015  11:13 AM           7778 Project_Readme.html
-a----        1/12/2016  10:25 AM           1178 Startup.cs
-a----        10/7/2015  11:13 AM           1213 WebApplication6.xproj
-a----        10/7/2015  11:43 AM            393 WebApplication6.xproj.user
```

**Comment by piotrpMSFT** [comment]
thanks @moozzyk 

I already moved these temporary folders int Temp in a non-merged PR. I'll let you know once that's merged.

**Comment by moozzyk** [comment]
:+1: 

(just curious, are they getting deleted from temp or still will be left behind?)

**Comment by piotrpMSFT** [comment]
They get deleted on the happy path, but the command has not been hardened against exception scenarios [yet].

**Comment by piotrpMSFT** [comment]
@Sridhar-MS fixed this in #877
